app/controllers/accounts_controller.rb

```diff
@@ -68,20 +68,27 @@ def get_profile
     if @current_user
       user = User.find_by(id: params[:id].to_i)
 
-      if not user
+      unless user
         render(inertia: 'NotFound')
         return
       end
 
+      user_json = user.as_json(only: [:id, :name, :username, :description, :avatar])
+
+      student_reviews = Review.where(connection_id: Connection.where(match_id: MatchRequest.where(student_id: user.id)))
+      teacher_reviews = Review.where(connection_id: Connection.where(match_id: MatchRequest.where(skill_class_id: SkillClass.where(teacher_id: user.id))))
+
+      total_reviews = student_reviews.or(teacher_reviews)
+
+      avg_rating = total_reviews.average(:rating).to_f.round(1) # only one decimal case
+
```

Comment:
this entire logic could be moved into a service. like ProfileStats

```ruby
stats = ProfileStats.new(user).get()
```

Reply:
Makes sense, so I did it